File: app/src/main/java/eu/sesma/dagger/ui/detail/DetailPresenter.java
package eu.sesma.dagger.ui.detail;

import eu.sesma.dagger.ui.ActCollaborator;
import eu.sesma.dagger.ui.Navigator;

import android.util.Log;

import javax.inject.Inject;

public class DetailPresenter {

    private static final String TAG = DetailPresenter.class.getSimpleName();

    private final Navigator navigator;
    private final ActCollaborator actColaborator;

    @Inject
    public DetailPresenter(Navigator navigator, ActCollaborator actColaborator) {
        this.navigator = navigator;
        this.actColaborator = actColaborator;
    }

    public void initialize() {

    }

    public void onClick() {
        showLogs();
        navigator.navigateToAnother();
    }

    private void showLogs() {
        Log.d(TAG, "ActCollaborator: " + actColaborator);
    }
}

File: app/src/main/java/eu/sesma/dagger/ui/detail/DetailPresenter.kt
package eu.sesma.dagger.ui.detail

import android.util.Log
import eu.sesma.dagger.ui.ActCollaborator
import eu.sesma.dagger.ui.Navigator
import javax.inject.Inject

class DetailPresenter
@Inject constructor(private val navigator: Navigator, private val actColaborator: ActCollaborator) {

    companion object {
        private val TAG = DetailPresenter::class.java.simpleName
    }

    fun initialize() {
    }

    fun onClick() {
        showLogs()
        navigator.navigateToAnother()
    }

    private fun showLogs() {
        Log.d(TAG, "ActCollaborator: $actColaborator")
    }
}
